File: python/py-wyrd/examples/from_yaml.py
"""Load a workflow definition from YAML and inspect its steps."""

from pathlib import Path

from wyrd import Workflow


def main() -> None:
    wf = Workflow.load(Path(__file__).parent / "workflows" / "research.yaml")
    print(f"workflow: {wf.name}")
    print(f"steps: {', '.join(wf.steps)}")


if __name__ == "__main__":
    main()

File: crates/skald/skald-workflow/examples/from_yaml.rs
//! Load a workflow from YAML and run it.

use std::collections::HashMap;

use skald_workflow::{Workflow, WorkflowInput};

fn main() -> anyhow::Result<()> {
    let wf = Workflow::load(
        "examples/workflows/research.yaml",
        skald_tool::default_registry(),
        skald_agent::default_prompt_resolver(),
    )?;
    let run = wyrd_runtime::runtime().block_on(wf.run(WorkflowInput::from(HashMap::from([(
        "topic".to_owned(),
        "climate change".to_owned(),
    )]))))?;
    println!("steps completed: {}", run.tasks.len());
    println!("parameters: {:?}", run.parameters);
    if let Some(text) = run.final_output.as_deref() {
        println!("final output: {text}");
    }
    Ok(())
}
